### ISSUE TITLE
[a11y][SelectDialog]: headerText is not read by VoiceOver

### ISSUE THREAD

**Issue by volha-arkhipenka-sap** [SAP/ui5-webcomponents#11474]
### Bug Description

With Voiceover enabled, on opening `SelectDialog` the title provided in `headerText` is not read by VoiceOver. 

### Affected Component

`SelectDialog` https://sap.github.io/ui5-webcomponents-react/v2/?path=/docs/modals-popovers-selectdialog--docs

### Expected Behaviour

All `SelectDialog` content and information should be conveyed to a user.

### Isolated Example

Reproducible with VoiceOver enabled on https://sap.github.io/ui5-webcomponents-react/v2/?path=/docs/modals-popovers-selectdialog--docs

### Steps to Reproduce

1. Provide `headerText` prop to `SelectDialog` component
2. Enable VoiceOver
3. Open `SelectDialog` component
4. `headerText` is not announced

### Log Output, Stack Trace or Screenshots

<img width="1724" alt="Image" src="https://github.com/user-attachments/assets/11371875-5d50-47a1-8f30-f8c000b1eb20" />

### Priority

None

### UI5 Web Components Version

2.9.0

### Browser

Chrome

### Operating System

macOS Sequoia 15.4.1

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by AlekseiSch** [comment]
Also reproducible for NVDA
![Image](https://github.com/user-attachments/assets/570010b5-e3b7-4fbf-bcdb-c8c442f0016e)

**Comment by NakataCode** [comment]
Hi @volha-arkhipenka-sap,

I tested the `<ui5-dialog>` component directly using the Playground with `header-text` provided, and observed that VoiceOver does read the header, as long as the initial focus is placed on it. You can check the test example [here](https://sap.github.io/ui5-webcomponents/play/#eyJpbmRleC5odG1sIjp7Im5hbWUiOiJpbmRleC5odG1sIiwiY29udGVudCI6IjwhLS0gcGxheWdyb3VuZC1mb2xkIC0tPlxuPCFET0NUWVBFIGh0bWw-XG48aHRtbCBsYW5nPVwiZW5cIj5cblxuXG5cblxuPGhlYWQ-XG4gICAgXG4gICAgPHN0eWxlPlxuICAgICAgKjpub3QoOmRlZmluZWQpIHtcbiAgICAgICAgZGlzcGxheTogbm9uZTtcbiAgICAgIH1cblxuICAgIGh0bWwge1xuICAgICAgZm9yY2VkLWNvbG9yLWFkanVzdDogbm9uZTtcbiAgICB9XG4gICAgPC9zdHlsZT5cblxuICAgIFxuICAgIDxzdHlsZT5cbiAgICAgICo6bm90KDpkZWZpbmVkKSB7XG4gICAgICAgIGRpc3BsYXk6IG5vbmU7XG4gICAgICB9XG5cbiAgICBodG1sIHtcbiAgICAgIGZvcmNlZC1jb2xvci1hZGp1c3Q6IG5vbmU7XG4gICAgfVxuICAgIDwvc3R5bGU-XG5cbiAgICBcbiAgICA8c3R5bGU-XG4gICAgICAqOm5vdCg6ZGVmaW5lZCkge1xuICAgICAgICBkaXNwbGF5OiBub25lO1xuICAgICAgfVxuXG4gICAgaHRtbCB7XG4gICAgICBmb3JjZWQtY29sb3ItYWRqdXN0OiBub25lO1xuICAgIH1cbiAgICA8L3N0eWxlPlxuXG4gICAgPG1ldGEgY2hhcnNldD1cIlVURi04XCI-XG4gICAgPG1ldGEgbmFtZT1cInZpZXdwb3J0XCIgY29udGVudD1cIndpZHRoPWRldmljZS13aWR0aCwgaW5pdGlhbC1zY2FsZT0xLjBcIj5cbiAgICA8dGl0bGU-U2FtcGxlPC90aXRsZT5cbiAgICA8bGluayByZWw9XCJzdHlsZXNoZWV0XCIgaHJlZj1cIi4vbWFpbi5jc3NcIj5cbjwvaGVhZD5cblxuPGJvZHkgc3R5bGU9XCJiYWNrZ3JvdW5kLWNvbG9yOiB2YXIoLS1zYXBCYWNrZ3JvdW5kQ29sb3IpOyBoZWlnaHQ6IDQ1MHB4O1wiPlxuPHVpNS1idXR0b24gaWQ9XCJyZXNpemFibGUtb3BlblwiPk9wZW4gRGlhbG9nPC91aTUtYnV0dG9uPlxuICAgIDwhLS0gcGxheWdyb3VuZC1mb2xkLWVuZCAtLT5cblxuICAgPHVpNS1kaWFsb2cgaWQ9XCJyZXNpemFibGUtZGlhbG9nXCIgaGVhZGVyLXRleHQ9XCJSZXNpemFibGUgZGlhbG9nIGhlYWRlclwiIHJlc2l6YWJsZT5cblx0XHQ8cD5IaW50OiBSZXNpemUgdGhpcyBkaWFsb2cgYnkgZHJhZ2dpbmcgaXQgYnkgaXRzIHJlc2l6ZSBoYW5kbGUgb3IgdXNlIHNoaWZ0K2Fycm93IGtleXMuPC9wPlxuXHRcdDxwPlRoaXMgZmVhdHVyZSBpcyBhdmFpbGFibGUgb25seSBvbiBEZXNrdG9wLjwvcD5cblx0XHQ8ZGl2IHNsb3Q9XCJmb290ZXJcIiBjbGFzcz1cImRpYWxvZ0Zvb3RlclwiPlxuXHRcdFx0PHVpNS1idXR0b24gaWQ9XCJyZXNpemFibGUtY2xvc2VcIiBkZXNpZ249XCJFbXBoYXNpemVkXCI-T0s8L3VpNS1idXR0b24-XG5cdFx0PC9kaXY-XG5cdDwvdWk1LWRpYWxvZz5cbiAgICA8IS0tIHBsYXlncm91bmQtZm9sZCAtLT5cbiAgICA8c2NyaXB0IHR5cGU9XCJtb2R1bGVcIiBzcmM9XCJtYWluLmpzXCI-PC9zY3JpcHQ-XG48L2JvZHk-XG5cbjwvaHRtbD5cbjwhLS0gcGxheWdyb3VuZC1mb2xkLWVuZCAtLT5cbiJ9LCJtYWluLmpzIjp7Im5hbWUiOiJtYWluLmpzIiwiY29udGVudCI6Ii8qIHBsYXlncm91bmQtaGlkZSAqL1xuaW1wb3J0IFwiLi9wbGF5Z3JvdW5kLXN1cHBvcnQuanNcIjtcbi8qIHBsYXlncm91bmQtaGlkZS1lbmQgKi9cbmltcG9ydCBcIkB1aTUvd2ViY29tcG9uZW50cy9kaXN0L0RpYWxvZy5qc1wiO1xuaW1wb3J0IFwiQHVpNS93ZWJjb21wb25lbnRzL2Rpc3QvQnV0dG9uLmpzXCI7XG5cblxud2luZG93W1wicmVzaXphYmxlLW9wZW5cIl0uYWRkRXZlbnRMaXN0ZW5lcihcImNsaWNrXCIsIGZ1bmN0aW9uICgpIHsgd2luZG93W1wicmVzaXphYmxlLWRpYWxvZ1wiXS5vcGVuID0gdHJ1ZTsgfSk7XG5cdFx0d2luZG93W1wicmVzaXphYmxlLWNsb3NlXCJdLmFkZEV2ZW50TGlzdGVuZXIoXCJjbGlja1wiLCBmdW5jdGlvbiAoKSB7IHdpbmRvd1tcInJlc2l6YWJsZS1kaWFsb2dcIl0ub3BlbiA9IGZhbHNlOyB9KTsifSwibWFpbi5jc3MiOnsibmFtZSI6Im1haW4uY3NzIiwiY29udGVudCI6Ii5sb2dpbi1mb3JtIHtcbiAgICBkaXNwbGF5OiBmbGV4O1xuICAgIGZsZXgtZGlyZWN0aW9uOiBjb2x1bW47XG4gICAganVzdGlmeS1jb250ZW50OiBzcGFjZS1ldmVubHk7XG4gICAgYWxpZ24taXRlbXM6IGZsZXgtc3RhcnQ7XG4gICAgbWFyZ2luOiAzcmVtIDZyZW07XG59XG5cbi5sb2dpbi1mb3JtID4gZGl2IHtcbiAgICBkaXNwbGF5OiBncmlkO1xuICAgIHdpZHRoOiAxNXJlbTtcbiAgICBtYXJnaW4tYm90dG9tOiAuNXJlbTtcbn1cblxuQG1lZGlhKG1heC13aWR0aDogNjAwcHgpIHtcbiAgICAubG9naW4tZm9ybSB7XG4gICAgICAgIG1hcmdpbjogM3JlbSAxcmVtO1xuICAgIH1cbn0ifX0).

According to the spec, the dialog title should be announced on focus, so the behavior depends on where the focus lands upon dialog open. In your example, the initial focus is set on the first item inside the dialog, not on the header, which likely explains why VoiceOver doesn’t announce it.

I’m going to proceed with closing this ticket for now. If any further questions arise or you need clarification, feel free to reopen it.

Best regards,
Nikola

**Comment by volha-arkhipenka-sap** [comment]
Hi @NakataCode, thank you for the response.

However, when `SelectDialog` contains `ListItemStandard`s, the focus is set on the first item automatically by `SelectDialog`. This focus behavoir seems to be default and internal for `SelectDialog` component when it has a list of `ListItemStandard`s inside. The focus lands on the first `ListItemStandard` even in the example from the documentation https://sap.github.io/ui5-webcomponents-react/v2/?path=/docs/modals-popovers-selectdialog--docs.

Should it be considered as a bug of incorrect focus landing on `SelectDialog` open?

**Comment by NakataCode** [comment]
Hello @SAP/ui5-webcomponents-topic-rd,

Could you please take over this issue?

Best regards,
Nikola

**Comment by LidiyaGeorgieva** [comment]
Hello @Lukas742,

It looks that this should be fixed in the Select Dialog, where slot "header" is used, but "accessibleName" is not provided.
See the documentation of the DIalog: https://sap.github.io/ui5-webcomponents/nightly/components/Dialog/#header

Best Regards,
Lidiya

**Comment by Lukas742** [comment]
Hi @LidiyaGeorgieva 

thanks for letting me know! I'll move this issue to our repo.

___

Hi @volha-arkhipenka-sap 

until an accessible name is available out of the box, you can set it yourself by setting the `accessibleName` prop.
